ui/app/components/app/transaction-list/transaction-list-hooks.component.js

```diff
@@ -0,0 +1,161 @@
+  const firstPendingTransactionId =
+    pendingTransactions[0] && pendingTransactions[0].primaryTransaction.id
+  return {
+    completedTransactions: nonceSortedCompletedTransactionsSelector(state),
+    pendingTransactions,
+    firstPendingTransactionId,
+    selectedToken: selectedTokenSelector(state),
+    selectedAddress: getSelectedAddress(state),
+    assetImages: getAssetImages(state),
+    transactionTimeFeatureActive: getFeatureFlags(state).transactionTime,
+  }
+}
+
+function useTransactionListState () {
+  const dispatch = useDispatch()
+  const {
+    completedTransactions,
+    firstPendingTransactionId,
+    transactionTimeFeatureActive,
+    assetImages,
+    pendingTransactions,
+    // selectedAddress,
+    selectedToken,
+  } = useSelector(mapStateToProps)
+  const { fetchGasEstimates, fetchBasicGasAndTimeEstimates } = useMemo(() => ({
+    fetchGasEstimates: (blockTime) => dispatch(actions.fetchGasEstimates(blockTime)),
+    fetchBasicGasAndTimeEstimates: () => dispatch(actions.fetchBasicGasAndTimeEstimates()),
+  }), [dispatch])
+  const prevState = useRef({ loaded: false, pendingTransactions, transactionTimeFeatureActive })
+  const { loaded } = prevState.current
+  const pendingTransactionAdded = pendingTransactions.length > 0 && prevState.current.pendingTransactions.length === 0
+  const transactionTimeFeatureWasActivated = !prevState.current.transactionTimeFeatureActive && transactionTimeFeatureActive
+  useEffect(() => {
+    if (transactionTimeFeatureActive && pendingTransactions.length > 0 && (loaded === false || transactionTimeFeatureWasActivated || pendingTransactionAdded)) {
+      fetchBasicGasAndTimeEstimates()
+        .then(({ blockTime }) => fetchGasEstimates(blockTime))
+    }
+    prevState.current = { loaded: true, pendingTransactions, transactionTimeFeatureActive }
+  }, [fetchGasEstimates, fetchBasicGasAndTimeEstimates, transactionTimeFeatureWasActivated, transactionTimeFeatureActive, pendingTransactionAdded, pendingTransactions ])
+  return {
+    completedTransactions,
+    firstPendingTransactionId,
+    assetImages,
+    pendingTransactions,
+    selectedToken,
+  }
+}
+
+
+export default function TransactionList ({ isWideViewport = false } = {}) {
+  const {
+    pendingTransactions,
+    completedTransactions,
+    firstPendingTransactionId,
+    selectedToken,
+    assetImages,
+  } = useTransactionListState()
+
+  const t = useI18nContext()
+
+  const pendingLength = pendingTransactions.length
+
+  const shouldShowCancel = useCallback((transactionGroup) => {
+    const { hasCancelled } = transactionGroup
+    return !hasCancelled
+  }, [])
+
+  const shouldShowSpeedUp = useCallback((transactionGroup, isEarliestNonce) => {
+    const { transactions = [], hasRetried } = transactionGroup
+    const [earliestTransaction = {}] = transactions
+    const { submittedTime } = earliestTransaction
+    return Date.now() - submittedTime > 5000 && isEarliestNonce && !hasRetried
+  }, [])
+
+  return (
```

Comment:
Prior to this change, the component was split up into a few different `renderX` calls. I've consolidated them in this flow. However, if we wanted to separate them out in the future my recommendation would be to create new components for those rather than functions.